src/session/content/message_row/media.rs

```diff
@@ -146,6 +161,24 @@ impl Media {
         glib::Object::new(&[]).expect("Failed to create Media")
     }
 
+    fn update_indicators(&self) {
+        let imp = self.imp();
+        if imp.caption_label.is_visible() {
+            let indicators = self.indicators();
+            if let Some(ref indicators) = indicators {
+                imp.overlay.remove_overlay(indicators);
```

Comment:
This gives a critical when an existing caption of the media is edited.

```
(telegrand:2): Gtk-CRITICAL **: 01:22:44.060: gtk_overlay_remove_overlay: assertion 'gtk_widget_get_parent (widget) == GTK_WIDGET (overlay)' failed
```

This function needs to be redone in a different way.

Reply:
The changes were a bit more involved, but at least the code is a bit better imho.